modules/ROOT/pages/dataweave-formats.adoc

```diff
@@ -62,6 +62,75 @@ DataWeave supports these formats (or MIME types) as input and output:
+and for some data formats, part-by-part by streaming the input. When attempting
+to read a large file, it is possible to run out of memory or to impact
+performance negatively. Streaming can improve performance but impacts access to file.
+
+* Indexed and In-Memory: Allow for random access to data because both
+strategies parse the entire document. For these strategies, your DataWeave
+script can access any part of the resulting value at any time.
+** Indexed: Uses indexes over the disk.
+** In-Memory: Parses the entire document in memory.
+
+* Streaming: Allows for sequential access to the file. This strategy partitions
+the input document into smaller items and accesses its data sequentially,
+storing the current item in memory. A DataWeave selector can access the portion
+of the file that is getting read. DataWeave supports streaming for a few
+formats:
+** <<format_csv, CSV>>
+** <<format_json, JSON>>
+
+[[reader_writer_properties]]
+=== Using Reader and Writer Properties
+
+In some cases, it is necessary to modify or specify aspects of the format
+through format-specific properties. For example, you can specify CSV input and
+output properties, such as the `separator` (or delimiter) to use in the CSV file.
+For Cobol copybook, you need to specify the path to a schema file using the
+`schemaPath` property.
+
+You can append reader properties to the MIME type (`outputMimeType`) attribute
+for certain components in your Mule app. Listeners and Read operations accept
+these settings. For example, this On New File listener example identifies the `,`
+separator for a CSV input file:
+
+.Example: Properties for the CSV Reader
+[source,xml,linenums]
+----
+<file:listener doc:name="On New File" config-ref="File_Config" outputMimeType='application/csv; separator=","'>
+  <scheduling-strategy >
+    <fixed-frequency frequency="45" timeUnit="SECONDS"/>
+  </scheduling-strategy>
+  <file:matcher filenamePattern="comma_separated.csv" />
+</file:listener>
+----
+
+Note that the `outputMimeType` setting above helps the CSV _reader_ interpret
+the format and delimiter of the input `comma_separated.csv` file, not the writer.
+
+To specify the output format, you can provide the MIME type and any writer
+properties for the writer, such as the CSV or JSON writer used by a File Write
+operation. For example, you might need to write a pipe (`|`) delimiter in your
+CSV output payload, instead of some other delimiter used in the input. To do
+this, you append the property and its value to the `output` directive of a
+DataWeave expression. For example, this Write operation specifies the pipe as
+a `separator`:
+
+.Example: output Directive for the CSV Writer
+[source,xml,linenums]
+----
+<file:write doc:name="Write" config-ref="File_Config" path="my_transform">
+  <file:content ><![CDATA[#[output application/csv separator="|" --- payload]]]></file:content>
+</file:write>
+----
+
+The sections below list the format-specific reader and writer properties
```

Comment:
Omit this type of transition between topics, which we want to be whole, independent units.
Consider using a See Also link instead.